opm/grid/GraphOfGridWrappers.cpp

```diff
@@ -208,201 +208,159 @@ void extendGIDtoRank(const GraphOfGrid<Dune::CpGrid>& gog,
-                    {
-                        commData.push_back(gID);
-                    }
-                }
-                assert(totsize==(int)commData.size());
-                int tag = 37; // a random number
-                MPI_Send(&totsize, 1, MPI_INT, i, tag++, cc);
-                MPI_Send(commData.data(), totsize, MPI_INT, i, tag, cc);
-            }
+            int ii=i+(int)(i>=root); // ii takes values {0,...,mpisize-1} but skips root
+            int tag = 37; // a random number
+            nrWells[i]=exportedCells[ii].size();
+            MPI_Isend(&nrWells[i], 1, MPI_INT, ii, tag++, cc, &requestSize[2*i]);
+            MPI_Isend(exportedCells[ii].data(), nrWells[i], MPI_INT, ii, tag++, cc, &requestSize[2*i+1]);
         }
+        // wait till all messages are completed
+        MPI_Waitall(requestSize.size(), requestSize.data(), MPI_STATUS_IGNORE);
     }
     else // receive data from root
     {
         int tag = 37; // a random number
-        int totsize;
-        MPI_Recv(&totsize, 1, MPI_INT, root, tag++, cc, MPI_STATUS_IGNORE);
-        std::vector<int> receivedData(totsize);
-        MPI_Recv(receivedData.data(), totsize, MPI_INT, root, tag, cc, MPI_STATUS_IGNORE);
-
-        int numWells = receivedData[0];
-        result.resize(numWells);
-        int index = 1;
-        for (int i=0; i<numWells; ++i)
-        {
-            int wellSize = receivedData[index++];
-            assert(index+wellSize<=totsize);
-            const auto dataBegin = receivedData.begin()+index;
-            result[i] = std::vector<int>(dataBegin, dataBegin+wellSize);
-            index+=wellSize;
-        }
+        int nrWells;
+        MPI_Recv(&nrWells, 1, MPI_INT, root, tag++, cc, MPI_STATUS_IGNORE);
+        result.resize(nrWells);
+        MPI_Recv(result.data(), nrWells, MPI_INT, root, tag++, cc, MPI_STATUS_IGNORE);
     }
     return result;
 }
 
-void extendImportList(std::vector<std::tuple<int,int,char,int>>& importList,
-                      const std::vector<std::vector<int>>& extraWells)
+
+void extendAndSortImportList(std::vector<std::tuple<int,int,char,int>>& importList,
+                             const std::vector<int>& extraWells)
 {
-    using ImportList = std::vector<std::tuple<int,int,char,int>>;
-    // make a list of wells for easy identification
-    std::unordered_map<int, std::size_t> wellMap;
-    for (std::size_t i=0; i<extraWells.size(); ++i)
+    if (importList.empty() || extraWells.empty())
     {
-        if (extraWells[i].size()>1)
-        {
-            wellMap[extraWells[i][0]] = i;
-        }
+        std::sort(importList.begin(), importList.end());
+        return;
     }
+    int rank = std::get<1>(importList[0]);
 
-    ImportList addToList;
-    // iterate once through the original importList
-    for (const auto& cellProperties : importList)
+    importList.reserve(importList.size()+extraWells.size());
+    for (const auto& well : extraWells)
```

Comment:
`well` does not denote a well here, but a cell, right? Please change the name to reflect this.